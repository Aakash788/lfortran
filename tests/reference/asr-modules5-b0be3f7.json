{
    "basename": "asr-modules5-b0be3f7",
    "cmd": "lfortran --indent --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/modules5.f90",
    "infile_hash": "19d649ba534a6988744fc82fc7299f5b95c3ef5213106f4a1ebf48e8",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-modules5-b0be3f7.stdout",
    "stdout_hash": "bc53b427dfa1f1f4730eff06b60910fd4a096ad7fef614cd6d6d6789",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}