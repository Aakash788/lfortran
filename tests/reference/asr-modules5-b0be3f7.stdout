(TranslationUnit
    (SymbolTable
        1
        {
            a_foo:
                (Module
                    (SymbolTable
                        4
                        {
                            check_proc:
                                (Function
                                    (SymbolTable
                                        7
                                        {
                                            array:
                                                (Variable
                                                    7
                                                    array
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Pointer
                                                        (Struct
                                                            4 child_value
                                                            []
                                                        )
                                                    )
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            ptr:
                                                (Variable
                                                    7
                                                    ptr
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Pointer
                                                        (Class
                                                            4 toml_value
                                                            []
                                                        )
                                                    )
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                ),
                                            ~select_type_block_:
                                                (Block
                                                    (SymbolTable
                                                        8
                                                        {
                                                            1_child_value_ok:
                                                                (ExternalSymbol
                                                                    8
                                                                    1_child_value_ok
                                                                    5 ok
                                                                    child_value
                                                                    []
                                                                    ok
                                                                    Public
                                                                )
                                                        })
                                                    ~select_type_block_
                                                    [(If
                                                        (StructInstanceMember
                                                            (Var 7 ptr)
                                                            8 1_child_value_ok
                                                            (Logical 4 [])
                                                            ()
                                                        )
                                                        [(=>
                                                            (Var 7 array)
                                                            (Var 7 ptr)
                                                        )]
                                                        [(SubroutineCall
                                                            4 not_ok
                                                            ()
                                                            []
                                                            ()
                                                        )
                                                        (Return)]
                                                    )]
                                                )
                                        })
                                    check_proc
                                    [not_ok
                                    default_class]
                                    []
                                    [(If
                                        (PointerAssociated
                                            (Var 7 ptr)
                                            ()
                                            (Logical 4 [])
                                            ()
                                        )
                                        [(SelectType
                                            [(TypeStmtName
                                                4 child_value
                                                [(BlockCall
                                                    -1
                                                    7 ~select_type_block_
                                                )]
                                            )]
                                            [(SubroutineCall
                                                4 default_class
                                                ()
                                                []
                                                ()
                                            )
                                            (Return)]
                                        )]
                                        []
                                    )]
                                    ()
                                    Source
                                    Public
                                    Implementation
                                    ()
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    []
                                    []
                                    .false.
                                ),
                            child_value:
                                (StructType
                                    (SymbolTable
                                        5
                                        {
                                            ok:
                                                (Variable
                                                    5
                                                    ok
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Logical 4 [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    child_value
                                    []
                                    [ok]
                                    Source
                                    Public
                                    .false.
                                    ()
                                    4 toml_value
                                ),
                            default_class:
                                (Function
                                    (SymbolTable
                                        9
                                        {
                                            
                                        })
                                    default_class
                                    []
                                    []
                                    [(Print
                                        ()
                                        [(StringConstant
                                            "default class"
                                            (Character 1 13 () [])
                                        )]
                                        ()
                                        ()
                                    )]
                                    ()
                                    Source
                                    Public
                                    Implementation
                                    ()
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    []
                                    []
                                    .false.
                                ),
                            new:
                                (GenericProcedure
                                    4
                                    new
                                    [4 check_proc]
                                    Public
                                ),
                            not_ok:
                                (Function
                                    (SymbolTable
                                        10
                                        {
                                            
                                        })
                                    not_ok
                                    []
                                    []
                                    [(Print
                                        ()
                                        [(StringConstant
                                            "not ok"
                                            (Character 1 6 () [])
                                        )]
                                        ()
                                        ()
                                    )]
                                    ()
                                    Source
                                    Public
                                    Implementation
                                    ()
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    []
                                    []
                                    .false.
                                ),
                            toml_value:
                                (StructType
                                    (SymbolTable
                                        6
                                        {
                                            key:
                                                (Variable
                                                    6
                                                    key
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Allocatable
                                                    (Character 1 -2 () [])
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                )
                                        })
                                    toml_value
                                    []
                                    [key]
                                    Source
                                    Public
                                    .false.
                                    ()
                                    ()
                                )
                        })
                    a_foo
                    []
                    .true.
                    .false.
                ),
            d_foo:
                (Module
                    (SymbolTable
                        2
                        {
                            check_here:
                                (Function
                                    (SymbolTable
                                        11
                                        {
                                            
                                        })
                                    check_here
                                    [check_proc]
                                    []
                                    [(SubroutineCall
                                        2 check_proc
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Source
                                    Public
                                    Implementation
                                    ()
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    .false.
                                    []
                                    []
                                    .false.
                                ),
                            check_proc:
                                (ExternalSymbol
                                    2
                                    check_proc
                                    4 check_proc
                                    a_foo
                                    []
                                    check_proc
                                    Public
                                )
                        })
                    d_foo
                    [a_foo]
                    .false.
                    .false.
                )
        })
    []
)
